### ISSUE TITLE
[CP] [macOS] Suppress Xcode 15 createItemModels warning

### ISSUE THREAD

**Issue by cbracken** [flutter/flutter#139284]
### Issue Link

https://github.com/flutter/flutter/issues/135277

### Commit Hash

d71fe92ec4b17b7d5bfdff1f44658f9c4a99fcc2

### Target

stable

### PR Link

https://github.com/flutter/flutter/pull/139782

### Changelog Description

Eliminates an excessive amount of Xcode error/warning output to the console when building or running macOS Flutter apps.

### Impacted Users

macOS developers

### Impact Description

Xcode 15 emits a large amount of internal Xcode warnings during the build process. These are filtered automatically on the Xcode console but not on stderr, which resulted in them being emitted to the terminal for builds from the command line or as part of CI scripts. Impact is that legitimate warnings/errors get lost in the noise of these errors, or misreported, leading to developers wasting time chasing red herrings.

### Workaround

No.

### Risk

low

### Test Coverage

yes

### Validation Steps

Added unit tests, manual validation.

**Comment by CaseyHillers** [comment]
@cbracken before we can go ahead with this, we'll need a PR opened to the stable branch. Can you update your PR description with a PR pointing to flutter-3.16-candidate.0? Thanks!

**Comment by cbracken** [comment]
@CaseyHillers I've opened a PR at https://github.com/flutter/flutter/pull/139782 against flutter-3.16-candidate.0 as requested.

**Comment by elkon028** [comment]
@cbracken 
```
> sw_vers
ProductName:		macOS
ProductVersion:		14.1.2
BuildVersion:		23B2091
```
```
> xcodebuild -version
Xcode 15.0.1
Build version 15A507
```
### I updated to 3.16.3
```
Flutter 3.16.3 • channel stable • https://github.com/flutter/flutter.git
Framework • revision b0366e0a3f (4 days ago) • 2023-12-05 19:46:39 -0800
Engine • revision 54a7145303
Tools • Dart 3.2.3 • DevTools 2.28.4
```
### The problem still persists （`debug mode`）
```
> flutter --version
Launching lib/main.dart on macOS in debug mode...
--- xcodebuild: WARNING: Using the first of multiple matching destinations:
{ platform:macOS, arch:arm64, id:00006034-000249300A78001C }
{ platform:macOS, arch:x86_64, id:00006034-000249300A78001C }
2023-12-10 10:48:48.225 xcodebuild[18416:1139576] [MT] DVTAssertions: Warning in /System/Volumes/Data/SWE/Apps/DT/BuildRoots/BuildRoot11/ActiveBuildRoot/Library/Caches/com.apple.xbs/Sources/IDEFrameworks/IDEFrameworks-22269/IDEFoundation/Provisioning/Capabilities Infrastructure/IDECapabilityQuerySelection.swift:103
Details:  createItemModels creation requirements should not create capability item model for a capability item model that already exists.
Function: createItemModels(for:itemModelSource:)
Thread:   <_NSMainThread: 0x142f05a60>{number = 1, name = main}
Please file a bug at https://feedbackassistant.apple.com with this warning message and any useful information you can provide.
```

**Comment by CaseyHillers** [comment]
Released in 3.16.4

**Comment by gdelgado11** [comment]
I have this [WARNING:flutter/common/graphics/persistent_cache.cc(284)] Could not acquire the persistent cache directory. Caching of GPU resources on disk is disabled.
